sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpression.cs

```diff
@@ -6,14 +6,17 @@
 using System.Linq;
 using System.Text.Json.Serialization;
 using System.Collections.Generic;
+using System.Reflection;
 
 namespace Azure.Core.Expressions.DataFactory
 {
     /// <summary>
     /// A class representing either a primitive value or an expression.
     /// For details on DataFactoryExpressions see https://learn.microsoft.com/en-us/azure/data-factory/control-flow-expression-language-functions#expressions.
     /// </summary>
-    /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="Array"/>, or <see cref="IList{TElement}"/>.</typeparam>
+    /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="TimeSpan"/>,
+    /// <see cref="DateTimeOffset"/>, <see cref="Uri"/>, <see cref="IList{String}"/>, <see cref="IList{TElement}"/>,
```

Comment:
Is there something special about `IList<string>` that isn't covered by `IList<TElement>`?

Reply:
Yeah TElement is actually for model types that have a JsonConverter defined.

Reply:
Will update.

Reply:
same with IDctionary<string, string>

Reply:
> nit: should it be `IList{string}` to match the cref to `string`?
> 
> Dotnet has always confused me on `String` vs `string`

This doesn't compile, for some reason it wants it to be String.